### PR TITLE
Deprecate csss-raspberry

### PR DESCRIPTION
**Pull request by micahdbak**
This PR moves all csss-raspberry code into a new src/shared folder to be used by all React apps in the CSSS frontend

### CODE REVIEW

**Comment by micahdbak** [comment]
@EarthenSky , all seems to have worked well merging :)

I'm going to update the README.md file to add some development information.

One thing I think deserves some thought is the naming of the folders in src, and how to distinguish apps from the shared folder.

Thoughts on my calling it "_shared" instead of "shared"? Or have the apps be capitalized?

File: package.json [hunk 1]
```diff
@@ -27,12 +32,13 @@
     "eslint-plugin-import": "^2.29.1",
     "eslint-plugin-n": "^15.7.0",
     "eslint-plugin-promise": "^6.1.1",
+>>>>>>> main
```
Comment (EarthenSky) [inline package.json:35]: This is a git merge diff you missed, should probably decide on which dependencies you need.

File: src/main/pages/Home.js [hunk 2]
```diff
@@ -18,6 +32,7 @@ export const Home = () => {
           <a href="https://sfucsss.org">
             Click here to go see the old site.
           </a>
+>>>>>>> main
```
Comment (EarthenSky) [inline src/main/pages/Home.js:35]: Another merge you missed.

File: src/main/pages/NotFound.js [hunk 3]
```diff
@@ -6,7 +6,12 @@ export const NotFound = () => {
     <Page>
       <div className="p-4">
         <p>
+<<<<<<< HEAD
+          Error 404; Page not found :(
+          <br />
+=======
           Error 404; Page not found :(<br />
+>>>>>>> main
```
Comment (EarthenSky) [inline src/main/pages/NotFound.js:14]: ditto

**Review by EarthenSky — CHANGES_REQUESTED** [review]
You missed a bunch of merge diffs. Please Ctrl+F `<<<<<<< HEAD`. I'll check again after they're fixed

File: package.json (hunk 1 above)
Comment (micahdbak) [reply package.json:35]: Oops

File: src/main/pages/Home.js (hunk 2 above)
Comment (micahdbak) [reply src/main/pages/Home.js:35]: Oooooops

File: src/main/pages/NotFound.js (hunk 3 above)
Comment (micahdbak) [reply src/main/pages/NotFound.js:14]: Ooooooooooooooops

**Comment by micahdbak** [comment]
@EarthenSky i'll do a full `grep` for the diffs, my bad

**Comment by micahdbak** [comment]
I'll do _shared! I like it too

**Comment by micahdbak** [comment]
@EarthenSky , check out the new [README](https://github.com/CSSS/csss-site-frontend/tree/mdb-shared-code?tab=readme-ov-file#csss-site-frontend)

File: src/_shared/VSCode/NavItem.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { useLocation, Link } from 'react-router-dom';
+
+import { File } from '../Icon';
+
+export const NavItem = ({ text, to, isInFolder }) => {
+  // TODO: isSelected should highlight an item
```
Comment (EarthenSky) [inline src/_shared/VSCode/NavItem.js:7]: make an issue for this please (so new people can work on it) & include the issue number in the comment

File: src/_shared/VSCode/NavLink.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+import * as Icon from '../Icon';
+
+export const NavLink = ({ text, href, isInFolder, isSelected }) => {
+  // TODO: isSelected should highlight an item
```
Comment (EarthenSky) [inline src/_shared/VSCode/NavLink.js:7]: ditto

File: src/_shared/helpers.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
+const csssApi = 'https://api.sfucsss.org';
```
Comment (EarthenSky) [inline src/_shared/helpers.js:1]: in js do we write constants in screaming snake case? CSSS_API

**Review by EarthenSky — CHANGES_REQUESTED** [review]

File: src/_shared/VSCode/NavItem.js (hunk 4 above)
Comment (micahdbak) [reply src/_shared/VSCode/NavItem.js:7]: Hehehehe. Remember how you said that comments can become outdated? This comment is outdated - this component already does this. I'll remove it - good catch!

File: src/_shared/VSCode/NavLink.js (hunk 5 above)
Comment (micahdbak) [reply src/_shared/VSCode/NavLink.js:7]: Ditto to you too!

File: src/_shared/helpers.js (hunk 6 above)
Comment (micahdbak) [reply src/_shared/helpers.js:1]: Eh, not often. `const` doesn't always mean the same thing as other languages btw, but in this case it does. I'll do a find and replace for this cuz i like screaming snake case

Comment (EarthenSky) [reply src/_shared/helpers.js]: I like this rule \:) https://javascript.info/variables#uppercase-constants:~:text=capital%2Dnamed%20constants%20are%20only%20used%20as%20aliases%20for%20%E2%80%9Chard%2Dcoded%E2%80%9D%20values

File: public/css/index.html [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<!doctype html>
+<html>
+  <head>
+    <title>Hacker!</title>
+  </head>
+  <body>
+    <p>Oh my! You've hacked us! Oh my oh my!!!</p>
+  </body>
+</html>
```
Comment (EarthenSky) [inline public/css/index.html:9]: why is this under the css folder?

**Review by EarthenSky — APPROVED** [review]
Looks good!

File: src/_shared/helpers.js (hunk 6 above)
Comment (micahdbak) [reply src/_shared/helpers.js]: yesss I like that too!!

Comment (micahdbak) [reply src/_shared/helpers.js]: I'll follow it from here on out then

File: public/css/index.html (hunk 7 above)
Comment (micahdbak) [reply public/css/index.html:9]: Cuz without it, git would delete the empty folder. But it's necessary for the compiled `tailwind.css` file that gets put in the `build` folder, so I thought i'd add something funny as a placeholder to keep the folder from being empty :)

**Comment by micahdbak** [comment]
Yay!!!

File: public/css/index.html (hunk 7 above)
Comment (EarthenSky) [reply public/css/index.html:9]: sounds good, but add a comment next time! haha

Comment (micahdbak) [reply public/css/index.html:9]: Aight will do!

**Pull request merged**